### ISSUE TITLE
Provide a base class for Error/Exception classes

### ISSUE THREAD

**Issue by linusthe3rd** [fastly/fastly-py#21]
When writing a try-except block to catch this library's exceptions, one has to add extra lines to catch everything:

```
    try:
        # make calls to fastly   
    except (
        fastly.errors.AuthenticationError,
        fastly.errors.InternalServerError,
        fastly.errors.BadRequestError,
        fastly.errors.NotFoundError
    ) as e:
        # handle exception
```

It would be nice if the error classes shared the same base class so that consumers only had to catch a single exception:

```
    try:
        # make calls to fastly   
    except fastly.errors.FastlyError as e:
        # handle exception
```

**Comment by lisac** [comment]
That shared base class is a nice idea, @strife25. I tried it in PR #28, will monitor it for feedback.

**Comment by BryanTheScott** [comment]
It would be great to see that merged back in to master.  I had the same reaction as linus, and the PR by @lisac seems painless enough to incorporate.

**Comment by psbanka** [comment]
Fixed with v0.2.3
